core/storage/trie/serialization/polkadot_codec.cpp

```diff
@@ -197,39 +219,39 @@ namespace kagome::storage::trie {
   outcome::result<void> PolkadotCodec::encodeValue(
       common::Buffer &out,
       const TrieNode &node,
-      StateVersion version,
-      const StoreChildren &store_children) const {
-    auto hash = node.value.hash;
+      std::optional<StateVersion> version,
+      const ChildVisitor &child_visitor) const {
+    auto hash = node.getValue().hash;
     if (shouldBeHashed(node, version)) {
-      hash = hash256(*node.value.value);
-      if (store_children) {
-        OUTCOME_TRY(store_children(nullptr, *hash, Buffer{*node.value.value}));
+      hash = hash256(*node.getValue().value);
+      if (child_visitor) {
+        OUTCOME_TRY(child_visitor(node, *hash, Buffer{*node.getValue().value}));
```

Comment:
Need to distinguish encoded node from raw value (maybe add `bool is_value` argument to lambda)
```suggestion
        OUTCOME_TRY(child_visitor(nullptr, *hash, Buffer{*node.getValue().value}));
```